File: backend/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Order extends Model
{
    use SoftDeletes;

    public const TYPE_BASRA = 'basra';
    public const TYPE_DELIVERY = 'delivery';

    public const STATUS_PENDING = 'pending';
    public const STATUS_CONFIRMED = 'confirmed';
    public const STATUS_PROCESSING = 'processing';
    public const STATUS_SHIPPED = 'shipped';
    public const STATUS_DELIVERED = 'delivered';
    public const STATUS_COMPLETED = 'completed';
    public const STATUS_CANCELLED = 'cancelled';
    public const STATUS_REFUNDED = 'refunded';

    protected $fillable = [
        'reference',
        'customer_id',
        'governorate_id',
        'branch_id',
        'type',
        'status',
        'payment_method',
        'payment_status',
        'subtotal',
        'discount',
        'shipping_fee',
        'installation_fee',
        'total',
        'customer_name',
        'customer_phone',
        'customer_email',
        'shipping_address',
        'tracking_number',
        'daftra_invoice_id',
        'daftra_invoice_url',
        'daftra_meta',
        'notes',
        'placed_at',
    ];

    protected $casts = [
        'subtotal' => 'decimal:2',
        'discount' => 'decimal:2',
        'shipping_fee' => 'decimal:2',
        'installation_fee' => 'decimal:2',
        'total' => 'decimal:2',
        'daftra_meta' => 'array',
        'placed_at' => 'datetime',
    ];

    public function customer(): BelongsTo
    {
        return $this->belongsTo(Customer::class);
    }

    public function governorate(): BelongsTo
    {
        return $this->belongsTo(Governorate::class);
    }

    public function branch(): BelongsTo
    {
        return $this->belongsTo(Branch::class);
    }

    public function items(): HasMany
    {
        return $this->hasMany(OrderItem::class);
    }

    public function statusLogs(): HasMany
    {
        return $this->hasMany(OrderStatusLog::class);
    }

    public function bookings(): HasMany
    {
        return $this->hasMany(Booking::class);
    }
}

File: flutter_app/lib/features/orders/models/order.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../../core/models/branch.dart';
import '../../../core/models/governorate.dart';

part 'order.freezed.dart';
part 'order.g.dart';

/// Mirrors `frontend/types/order.type.ts` (`IOrder`).
@freezed
abstract class Order with _$Order {
  const factory Order({
    required int id,
    required String reference,
    required String type, // "delivery" | "basra"
    required String status,
    required String payment_method,
    required String payment_status,
    required num subtotal,
    required num discount,
    required num shipping_fee,
    required num installation_fee,
    required num total,
    required String customer_name,
    required String customer_phone,
    required String customer_email,
    required String shipping_address,
    required String tracking_number,
    required String placed_at,
    Governorate? governorate,
    Branch? branch,
    @Default(<OrderItem>[]) List<OrderItem> items,
  }) = _Order;

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
}

/// Mirrors `Item` nested in `order.type.ts`.
@freezed
abstract class OrderItem with _$OrderItem {
  const factory OrderItem({
    required int id,
    required int product_id,
    required String product_name,
    required String product_sku,
    required int quantity,
    required num unit_price,
    required num total,
  }) = _OrderItem;

  factory OrderItem.fromJson(Map<String, dynamic> json) =>
      _$OrderItemFromJson(json);
}
